### PR TITLE
Remove indirect build dependency on go-license-detector.

### PR DESCRIPTION
**Pull request by scunningham**
## What does this PR do?

Tweaks the dependencies-report script to explicitly download required version of the go-license-detector tool and invoke it directly.   Previous version expected the dependencies to be included in the go module files that it was analyzing.  This broke when newer version of 'go tidy' more aggressively strips modules that are not part of the targeted program's dependency tree.

## Why is it important?

The release build process invokes the 'dependencies-report' script, which is currently broken.

## Checklist

- [x ] My code follows the style guidelines of this project
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] I have made corresponding change to the default configuration files
- [x ] I have added tests that prove my fix is effective or that my feature works
- [x ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by michel-laterman** [comment]
If we want to take this approach we'll also need the go-licenser installed in a similar manner.
Maybe we should have a make command that will install both tools?

**Comment by scunningham** [comment]
Open to suggestions.  I am having similar issues with the licenser.

File: Makefile [hunk 1]
```diff
@@ -42,9 +45,9 @@ clean: ## - Clean up build artifacts
 .PHONY: generate
 generate: ## - Generate schema models
 	@printf "${CMD_COLOR_ON} Installing module for go generate\n${CMD_COLOR_OFF}"
-	go install github.com/aleksmaus/generate/...
+	env GOBIN=${GOBIN} go install github.com/aleksmaus/generate/cmd/schema-generate@latest
```
Comment (EricDavisX) [inline Makefile:48]: is it in the plan prior to 7.14 to remove alexmaus personal git directory as part of source?

Comment (scunningham) [reply Makefile:48]: That would be wise.  @aleksmaus can handle that when he returns.  Also noticed a [tool](https://github.com/elastic/fleet-server/blob/master/dev-tools/common.bash#L43) in Andrew Kroh's personal space.  Not sure where that came from.

**Comment by EricDavisX** [comment]
> Open to suggestions. I am having similar issues with the licenser.

If it works as is, I prefer to merge and iterate since it is blocking the build for all erryone

**Comment by scunningham** [comment]
> If it works as is, I prefer to merge and iterate since it is blocking the build for all erryone

Just waiting for a review.  Anybody?  Beuller?

File: Makefile (hunk 1 above)
Comment (aleksmaus) [reply Makefile:48]: This was a fork and heavily customized schema code generator for our specific needs.
FWIW, we pull bunch of third party dependencies from other people personal spaces ;-)

Can move it at some point into elastic org sure.

**Review by aleksmaus — APPROVED** [review]
LGTM

**Pull request merged**